docs/data-management/edf-to-bids.md

```diff
@@ -0,0 +1,634 @@
+
+
+!CAL Prenormalize: offx, offy = -18.822 -42.021
+
+
+!CAL Quadrant center: centx, centy =
+   5.882e-05  4.4115e-05
+
+
+!CAL Corner correction:
+   3.207e-05, -1.7627e-06
+  -1.6438e-05,  3.5065e-05
+  -2.0184e-05, -1.0586e-05
+   5.7616e-06,  1.5672e-05
+
+
+!CAL Gains: cx:120.898 lx:115.734 rx:103.799
+
+!CAL Gains: cy:164.746 ty:163.045 by:164.277
+
+!CAL Resolution (upd) at screen center: X=2.2, Y=1.6
+
+!CAL Gain Change Proportion: X: 0.115 Y: 0.008
+
+!CAL Gain Ratio (Gy/Gx) = 1.363
+
+!CAL Cross-Gain Ratios: X=0.100, Y=0.031
+
+!CAL Quadrant fixup[0] = 0.059,0.003
+
+!CAL Quadrant fixup[1] = 0.029,0.062
+
+!CAL Quadrant fixup[2] = 0.037,0.019
+
+!CAL Quadrant fixup[3] = 0.010,0.027
+
+!CAL PCR gain ratio(x,y) = 2.442, 1.946
+
+!CAL CR gain match(x,y) = 1.020, 1.020
+
+!CAL Slip rotation correction OFF
+
+!CAL CALIBRATION HV9 R RIGHT   GOOD
+
+!CAL VALIDATION HV9 R RIGHT GOOD ERROR 0.49 avg. 0.92 max  OFFSET 0.17 deg. -2.4,-3.7 pix.
+```
+
+The `!CAL` messages are parsed as follows:
+
+```Python
+# Parse calibration metadata
+metadata["CalibrationCount"] = 0
+if not calibration.empty:
+    warn("Calibration of more than one eye is not implemented")
+    calibration.trialid = calibration.trialid.str.replace("!CAL", "")
+    calibration.trialid = calibration.trialid.str.strip()
+
+    metadata["CalibrationLog"] = list(
+        zip(
+            calibration.trialid_time.values.astype(int),
+            calibration.trialid.values,
+        )
+    )
+
+    calibrations_msg = calibration.trialid.str.startswith(
+        "VALIDATION"
+    ) & calibration.trialid.str.contains("ERROR")
+    metadata["CalibrationCount"] = calibrations_msg.sum()
+
+    calibration_last = calibration.index[calibrations_msg][-1]
+    try:
+        meta_calib = re.match(
+            r"VALIDATION (?P<ctype>[\w\d]+) (?P<eyeid>[RL]+) (?P<eye>RIGHT|LEFT) "
+            r"(?P<result>\w+) ERROR (?P<avg>-?\d+\.\d+) avg\. (?P<max>-?\d+\.\d+) max\s+"
+            r"OFFSET (?P<offsetdeg>-?\d+\.\d+) deg\. "
+            r"(?P<offsetxpix>-?\d+\.\d+),(?P<offsetypix>-?\d+\.\d+) pix\.",
+            calibration.loc[calibration_last, "trialid"].strip(),
+        ).groupdict()
+
+        metadata["CalibrationType"] = meta_calib["ctype"]
+        metadata["AverageCalibrationError"] = [float(meta_calib["avg"])]
+        metadata["MaximalCalibrationError"] = [float(meta_calib["max"])]
+        metadata["CalibrationResultQuality"] = [meta_calib["result"]]
+        metadata["CalibrationResultOffset"] = [
+            float(meta_calib["offsetdeg"]),
+            (float(meta_calib["offsetxpix"]), float(meta_calib["offsetypix"])),
+        ]
+        metadata["CalibrationResultOffsetUnits"] = ["deg", "pixels"]
+    except AttributeError:
+        warn("Calibration data found but unsuccessfully parsed for results")
+```
+
+### Parsing the ``events`` dataframe
+
+The events dataframe contains several parameters calculated by the EyeLink software.
```

Comment:
Perhaps we should clarify that these events aren't equivalent to the ones in events.tsv or use a different name for them?

Reply:
YES! - please send a suggestion

Reply:
@esavary what about something like this?

```suggestion
The events dataframe contains several parameters calculated by the EyeLink software.
It is important to remark that these *events* are not exactly the same as the experimental *events* that are encoded in the dataset.
These *events* are specific to the recording of ET data with the EyeLink device.
```

Reply:
Yes, it looks good to me.